rules/aip0132/resource_reference_type.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aip0132
+
+import (
+	"github.com/googleapis/api-linter/lint"
+	"github.com/googleapis/api-linter/locations"
+	"github.com/googleapis/api-linter/rules/internal/utils"
+	"github.com/jhump/protoreflect/desc"
+)
+
+// List methods should reference the target resource via `child_type` or the
+// parent directly via `type`.
+var resourceReferenceType = &lint.MethodRule{
+	Name: lint.NewRuleName(132, "resource-reference-type"),
+	OnlyIf: func(m *desc.MethodDescriptor) bool {
+		repeated := utils.GetRepeatedMessageFields(m.GetOutputType())
```

Comment:
afaict, you are kinda bundling in a check that the OutputType, i.e. the Response Message, has a repeated field that, purportedly, is the repeated resources. However, this is not the end goal of this `MethodRule` afaict, but rather the goal is to check for the correct value of `type` or `child_type` on `resource_reference`.

In turn, we are not really doing a good job of the "bundled in" check, because we don't validate that the repeated field name is the plural of the resource type. In turn, I would remove this line and the check for `len(repeated) > 0`, and rather create a different `rules/aip0132/<name>.go` file with a `*lint.MethodRule` of its own that checks not only that there is *a* repeated field, but also that, that repeated field **is correct**, in terms of name and type, and all the things.

Make sense?

Reply:
> kinda bundling in a check

Yes and no. My intention here is to not _lint_ the response message for the necessary fields, but just ignore those that aren't in scope of this rule. The `resource-reference-type` rule shouldn't emit a Problem when the `repeated Resource` in the response doesn't actually have a `google.api.resource` annotation for some reason or if it is a primitive (which isn't technically allowed, but I've seen it). I'd rather keep this check for that reason - so that this rule doesn't warn on things that are out of its scope.

> we are not really doing a good job of the "bundled in" check...rather create a different rules/aip0132/<name>.go file with a *lint.MethodRule of its own that checks not only that there is a repeated field, but also that, that repeated field is correct, in terms of name and type, and all the things.

Yeah, totally, we should add these rules.